ipfs_stac/client.py

```diff
@@ -297,7 +328,7 @@ def __init__(
         self.cid = cid
         self.local_gateway = local_gateway
         self.api_port = api_port
-        self.data = None
+        self.data: bytes = None
```

Comment:
I'm not sure if this was on purpose but +1 for adding some static typing.  

This library desperately needs some static-typed love. That would have helped identify the bytes and bytesIO types that were interchangeably used.